source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,272 @@
+=============================
+Implementing DataManipulators
+=============================
+
+This is a guide for contributors who want to help with Data API implementation by creating DataManipulators.
+An updated list of DataManipulators to be implemented can be found at
+`SpongeCommon Issue #8 <https://github.com/SpongePowered/SpongeCommon/issues/8>`_.
+
+To fully implement a ``DataManipulator`` these steps must be followed:
+
+1. Implement the ``DataManipulator`` itself
+#. Implement the ``ImmutableDataManipulator``
+
+When these steps are complete, the following must also be done:
+
+3. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+#. Register everything in ``SpongeGameRegistry``
+
+.. note::
+  Don't forget to add an empty line at the end of every file you create.
+
+1. Implementing the DataManipulator
+===================================
+
+Implementing the ``DataManipulator`` is done by extending ``AbstractData`` in the easiest case. Let's assume we have a
+single value based manipulator, eg. the ``flying`` state on an entity, which can be ``true`` or ``false``.
+We decide to abstract things, so now we're able to just extend ``AbstractBooleanData``.
+Here is a short example:
+
+.. code-block:: java
+
+  public class SpongeFlyingData extends AbstractBooleanData<FlyingData, ImmutableFlyingData> implements FlyingData {
+ /* your code here */
+  }
+
+The next step is to provide the generic ``DataManipulator`` that follows the generic of ``FlyingData``. In this case
+we use ``FlyingData.class`` in our ``super`` constructor, followed by the ``Key`` which identifies the recommended
+``value`` of the flying state and last but not least the ``Key`` itself: ``Keys.IS_FLYING``.
+
+.. code-block:: java
+
+ public SpongeFlyingData(boolean flying) {
+          super(FlyingData.class, flying, Keys.IS_FLYING);
+ }
+
+In most cases while implementing an abstract Manipulator you want to have two constructors:
+
+* One without arguments (no-args) which calls the second
+* and one single-argument constructor which is actually used for the provided ``value``
+
+Now, after the Manipulator is done you have to check if you are extending ``AbstractData`` or not.
+Remember that we decided to extend ``AbstractBooleanData`` above.
+If you did too, you must now call ``registerFieldGetter()`` and ``registerFieldSetter`` in your constructor.
```

Comment:
I missed some brackets there. It should be ``registerFieldSetter()`` indicating it's a method. It does take arguments as shown in the code snippet.

If i understood this correctly you must use ``registerFieldSetter`` and ``registerFieldGetter`` when using an abstract class, but it's not mandatory when *not* using an abstract class.
@gabizou might explain this better.